### PR TITLE
[BD-46] docs: add 'Responsive' page

### PR DESCRIPTION
**Pull request by viktorrusakov**
## Description

Implement single source of truth for breakpoints used in Paragon, this is the improved version of [reverted PR](https://github.com/openedx/paragon/pull/1242) that fixes issues with MFE's

JIRA: [PAR-813](https://openedx.atlassian.net/browse/PAR-813)

### Deploy Preview

https://deploy-preview-1288--paragon-openedx.netlify.app/foundations/responsive

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @viktorrusakov! I've created [BLENDED-1232](https://openedx.atlassian.net/browse/BLENDED-1232) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


When this pull request is ready, tag your edX technical lead.

**Comment by viktorrusakov** [comment]
@adamstankiewicz not sure if we need to merge this actually... I believe with this implementation consumers won't be able to override breakpoints used in `.js` files because they are always imported from Paragon's `.scss` file. I don't think there is a way to fix it 🤔 Sadly we can't use scss file as a single source of truth it seems 

Another idea I have is to create a context for breakpoints and export if from Paragon so consumers could override those values in all component they use in their apps, this would require them to override both css value and values used in component but I don't think there is another way
What do you think?

File: www/gatsby-config.js [hunk 1]
```diff
@@ -13,7 +13,16 @@ const segmentPlugin = {
 };
 
 const plugins = [
-  "gatsby-plugin-sass",
+  {
+    resolve: `gatsby-plugin-sass`,
+    options: {
+     cssLoaderOptions: {
+       modules: {
+         namedExport: false,
```
Comment (adamstankiewicz) [inline www/gatsby-config.js:21]: A'ha!  🪄

**Review by adamstankiewicz — APPROVED** [review]
LGTM! 🚀

**Comment by openedx-webhooks** [comment]
@viktorrusakov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1288&entry.1484655318=https://github.com/viktorrusakov&entry.752974735=2022-05-18+08:16&entry.1917517419=2022-05-20+10:44&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by adamstankiewicz** [comment]
> @adamstankiewicz not sure if we need to merge this actually... I believe with this implementation consumers won't be able to override breakpoints used in `.js` files because they are always imported from Paragon's `.scss` file. I don't think there is a way to fix it 🤔 Sadly we can't use scss file as a single source of truth it seems
> 
> Another idea I have is to create a context for breakpoints and export if from Paragon so consumers could override those values in all component they use in their apps, this would require them to override both css value and values used in component but I don't think there is another way What do you think?

@viktorrusakov Missed this comment before I merged, my bad! 🙃 

As we discussed in our meeting though, we agreed this is likely an issue but are going to defer on it for the time being as the current state is that consumers already cannot override the `breakpoints.js` import from Paragon in a custom SCSS theme. I filed a new issue that we can prioritize down the road to investigate a solution to allow custom themes to override the values in the `breakpoints.js` import, or a different solution altogether.

**Pull request merged**